### PR TITLE
Docs and block-template improvements

### PR DESCRIPTION
**Pull request by CiaranMn**
Makes a few improvements to the docs and the `block-template` based on community feedback:
- adds `image` and `examples` to `block-template` and the docs, and provide a fallback for blocks missing `image` (making good on [this comment](https://github.com/blockprotocol/blockprotocol/pull/183#issuecomment-1028969302))
- links specific discussions on topics where they have been opened (styling and security)
- quick start guide: clean up confusing language around "web components", change from "suggest" writing in React to "can" write in React

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201763438761199
  - https://app.asana.com/0/0/1201763438761195
  - https://app.asana.com/0/0/1201771822111973

### CODE REVIEW

**Review by kachkaev — APPROVED** [review]

File: packages/block-template/package.json
Comment (kachkaev) [inline packages/block-template/package.json:3]: We’ll need to publish the package after merging. Will you do this or shall I?

File: site/src/_pages/docs/1_developing-blocks.mdx
Comment (nonparibus) [inline site/src/_pages/docs/1_developing-blocks.mdx:11]: I think we still haven't cracked the first-sentence's wording. Is a divider block really providing **functionality**, or is it a stylistic element introduced to improve visual comprehension and/or symbolize separation?

```suggestion
Blocks are discrete components that render information or provide functionality. You can create them using any web technology or framework.
```

File: site/src/_pages/docs/1_developing-blocks.mdx
Comment (nonparibus) [inline site/src/_pages/docs/1_developing-blocks.mdx:12]:
```suggestion
In most of our examples, we use _React_, a ready-made library for writing components. If you need it, here’s [an intro to React](https://reactjs.org/docs/getting-started.html).
```

File: site/src/_pages/docs/1_developing-blocks.mdx
Comment (nonparibus) [inline site/src/_pages/docs/1_developing-blocks.mdx:14]:
```suggestion
Note: you can define blocks in other ways and aren't limited to using React. Although the docs currently only describe how to create blocks using React, we'll be adding other examples soon.
```

**Review by nonparibus — COMMENTED** [review]
Suggested changes - good to merge after these.

File: packages/block-template/package.json
Comment (CiaranMn) [reply packages/block-template/package.json:3]: I shall do it

**Review by nonparibus — APPROVED** [review]

**Review by kachkaev — APPROVED** [review]

**Pull request merged**
